build-system/tasks/create-module-compatible-scripts/index.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $$ = require('gulp-load-plugins')();
+const colors = require('ansi-colors');
+const gulp = $$.help(require('gulp'));
+const log = require('fancy-log');
+const MagicString = require('magic-string');
+const parser = require('@babel/parser');
+const through = require('through2');
+const traverse = require('@babel/traverse').default;
+
+/**
+ * Takes the file given by gulp and parses its ast to find ternary condition,
+ * returning either `window.global` or `this`.
+ * In order to make the script compatible with `<script type=module`
+ * we replace the `this` found in the condition with `self`.
+ * @param {Buffer|string} file
+ * @param {string=} encoding
+ * @param {function(Error, Object)} callback - Call this function (optionally with an
+ * error argument and data) when you are done processing the supplied chunk.
+ */
+function transform(file, encoding, callback) {
+  const code = file.contents.toString('utf8');
+  const ast = parser.parse(code);
+  const magicString = new MagicString(code);
+  const changes = [];
+
+  traverse(ast, {
```

Comment:
I don't understand why this wasn't implemented as a standard babel plugin.

Reply:
I am not 100% familiar with the terminology so I dont know what do you mean by that? 
Are you saying that we wont need the magic string if we just re-run this file through babel?

Reply:
Yes. Use babel to mutate the AST.

Reply:
Actually I agree, a standard babel plugin will keep the codebase more streamlined to read through